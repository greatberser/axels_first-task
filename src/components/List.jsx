import { List, ListItem, ListItemText } from '@mui/material';

const expensesList = [
  { id: 1, date: '2025-06-01', category: 'Food', amount: 50 },
  { id: 2, date: '2025-06-02', category: 'Transport', amount: 20 },
  { id: 3, date: '2025-06-03', category: 'Entertainment', amount: 30 },
];

export const ExpenseList = ({ expenses = expensesList }) => {
  return (
    <List>
      {expenses.map((expense) => (
        <ListItem sx={{ color: 'black' }} key={expense.id} divider>
          <ListItemText
            primary={`${expense.date} - ${expense.category}`}
            secondary={`$${expense.amount.toFixed(2)}`}
          />
        </ListItem>
      ))}
    </List>
  );
};

export default ExpenseList;
