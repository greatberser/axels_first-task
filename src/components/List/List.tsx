import { List, ListItem, ListItemText, Typography } from '@mui/material';
import type { Expense } from '../../store/ducks/expenses';
import {
  StyledItem,
  LeftSide,
  Category,
  DateText,
  AmountText,
  EmptyMess,
} from '../../styled/List';

type ExpenseListProps = {
  expenses: Expense[];
};

const ExpenseList: React.FC<ExpenseListProps> = ({ expenses }) => {
  if (!expenses || expenses.length === 0) {
    return <EmptyMess>No found</EmptyMess>;
  }
  return (
    <List>
      {expenses.map((expense) => (
        <StyledItem key={expense.id} divider>
          <LeftSide>
            <Category>{expense.category}</Category>
            <DateText>{expense.date}</DateText>
          </LeftSide>
          <AmountText>{Number(expense.amount).toFixed(2)}</AmountText>
        </StyledItem>
      ))}
    </List>
  );
};

export default ExpenseList;
