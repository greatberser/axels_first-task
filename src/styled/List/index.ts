import styled from 'styled-components';
import { ListItem, ListItemText, Typography } from '@mui/material';
import { colors, fontSizes } from '../theme';

export const EmptyMess = styled.h4`
  text-align: center;
  font-weight: normal;
  font-size: ${fontSizes.large};
  margin-top: 2rem;
`;

export const StyledItem = styled(ListItem)`
  && {
    background-color: ${colors.backgroundItem};
    margin: 0.5rem 0;
    border-radius: 8px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    &:hover {
      background-color: ${colors.hover};
    }
  }
`;

export const LeftSide = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const Category = styled(ListItemText)`
  && {
    font-weight: bold;
    color: #333;
    font-size: ${fontSizes.medium};
  }
`;

export const DateText = styled(Typography)`
  && {
    font-size: ${fontSizes.small};
    color: #666;
    margin-right: 1rem;
  }
`;

export const AmountText = styled.span`
  font-weight: bold;
  color: ${colors.success};
  font-size: ${fontSizes.large};
  margin-left: auto;
  padding: 0.5rem 1rem;
`;
