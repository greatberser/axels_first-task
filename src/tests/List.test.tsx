import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import List from '../components/List';

describe('List', () => {
  it('should match a snapshot', () => {
    const { asFragment } = render(<List expenses={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render empty message when no expenses', () => {
    render(<List expenses={[]} />);
    expect(screen.getByText(/no found/i)).toBeInTheDocument();
  });

  it('should render Food expenses', () => {
    const expenses = [
      { id: 1, category: 'Food', date: '2023-10-01', amount: 10.0 },
      { id: 2, category: 'Transport', date: '2023-10-02', amount: 20.0 },
    ];
    render(<List expenses={expenses} />);

    expect(screen.getByText(/food/i)).toBeInTheDocument();
  });

  it('should render Transport expenses', () => {
    const expenses = [
      { id: 1, category: 'Food', date: '2023-10-01', amount: 10.0 },
      { id: 2, category: 'Transport', date: '2023-10-02', amount: 20.0 },
    ];
    render(<List expenses={expenses} />);

    expect(screen.getByText(/transport/i)).toBeInTheDocument();
  });

  it('should render amount 10.0', () => {
    const expenses = [
      { id: 1, category: 'Food', date: '2023-10-01', amount: 10.0 },
      { id: 2, category: 'Transport', date: '2023-10-02', amount: 20.0 },
    ];
    render(<List expenses={expenses} />);

    expect(screen.getByText(/10\.00/i)).toBeInTheDocument();
    expect(screen.getByText(/20\.00/i)).toBeInTheDocument();
  });

  it('should render amount 20.0', () => {
    const expenses = [
      { id: 1, category: 'Food', date: '2023-10-01', amount: 10.0 },
      { id: 2, category: 'Transport', date: '2023-10-02', amount: 20.0 },
    ];
    render(<List expenses={expenses} />);

    expect(screen.getByText(/20\.00/i)).toBeInTheDocument();
  });
});
