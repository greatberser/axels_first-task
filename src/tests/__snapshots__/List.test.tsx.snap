// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`List should match a snapshot 1`] = `
<DocumentFragment>
  <h4
    class="sc-bRKDuR dIHunt"
  >
    No found
  </h4>
</DocumentFragment>
`;
